src/renderer/components/helpers/ChatMethods.js

```diff
@@ -50,8 +50,9 @@ export function openMapDialog (screenContext, selectedChat) {
   screenContext.openDialog('MapDialog', { selectedChat })
 }
 
-export async function openViewProfileDialog (chatid, screenContext) {
-  const fullChat = await callDcMethodAsync('chatList.getFullChatById', chatid)
-  const contact = fullChat.contacts[0]
+export async function openViewProfileDialog (screenContext, contact) {
+  if(Number.isInteger(contact)) {
+    contact = await callDcMethodAsync('chatList.getContact', contact)
+  }
```

Comment:
I would just always re-fetch the contact in this case.

Reply:
I dont think we need to do this. We need to update the selectedChat anyways if a contact changes, and if we already have it why refetch it?